From 722c294625eb1d06b6ce74bd6efb24ff461d2af9 Mon Sep 17 00:00:00 2001
From: Wencai You <wencai.you@amlogic.com>
Date: Mon, 25 Feb 2019 20:18:23 +0800
Subject: [PATCH] camera : camera feture add application for gtvs [1/3]

PD#SWPL-none

Problem:
add camera feature
cts testCameraManagerGetDeviceIdList fail
camera2 cannot open

Solution:
camera2 just open front add back camera
if no front camera
return external camera for app

Verify:
verify by ampere

Change-Id: Icc5bd457e842fe0c6998b594cbba38a584ed9568
Signed-off-by: Wencai You <wencai.you@amlogic.com>
---
 src/com/android/camera/one/v2/Camera2OneCameraManagerImpl.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/one/v2/Camera2OneCameraManagerImpl.java b/src/com/android/camera/one/v2/Camera2OneCameraManagerImpl.java
index a4b5a9b..c0e6c04 100644
--- a/src/com/android/camera/one/v2/Camera2OneCameraManagerImpl.java
+++ b/src/com/android/camera/one/v2/Camera2OneCameraManagerImpl.java
@@ -145,6 +145,8 @@ public class Camera2OneCameraManagerImpl implements OneCameraManager {
         String cameraId = findFirstCameraIdFacing(CameraCharacteristics.LENS_FACING_FRONT);
         if (cameraId == null) {
             Log.w(TAG, "No front-facing camera found.");
+            Log.w(TAG, "use usb external camera for front-facing");
+            cameraId = findFirstCameraIdFacing(CameraCharacteristics.LENS_FACING_EXTERNAL);
         }
         return cameraId;
     }
@@ -167,4 +169,4 @@ public class Camera2OneCameraManagerImpl implements OneCameraManager {
         return null;
     }

-}
\ No newline at end of file
+}

1.9.1

